src/main/java/dan200/computercraft/core/apis/TermAPI.java

```diff
@@ -101,6 +111,59 @@ public static int parseColour( Object[] args, boolean _enableColours ) throws Lu
         };
     }
 
+    public static void setColour( Terminal terminal, int colour, float r, float g, float b )
+    {
+        if( terminal.getPalette() != null )
+        {
+            terminal.getPalette().setColour( colour, r, g, b );
+            terminal.setChanged();
+        }
+    }
+
+    public static void setColour( Terminal terminal, HashMap<Object, Object> colours) throws LuaException
```

Comment:
To quote @BombBloke:

> [...] Otherwise, trying to change all 16 colours in a window (and how often will people change one colour at a time?) requires 16 pairs loops, and hence 256 calls to the parent terminal's setColour function, and if that parent has a parent, it only gets worse from there.

I agree that the overload adds clutter, so I think we should benchmark this to determine if it's really needed.